core/conn/jdbc_type2/native/SqlInterface.cpp

```diff
@@ -246,10 +246,14 @@ SQLRETURN GetJDBCValues( SQLItemDesc_def *SQLItemDesc,
 		SQLItemDesc->signType      = TRUE;
 		break;
 	case SQLTYPECODE_DATETIME:
+		// Calculate the length based on YYYY/MM/DD HH:MM:SS.ffffff
+        SQLItemDesc->maxLen = 20 + 6;
```

Comment:
SQLItemDesc->maxLen is used as input for some of the data types. Does this mean it is invalid for SQLItemDesc for date columns. If so, you might consider fixing where SQLItemDesc is obtained from SQL.